Sources/Core/EthereumABI/ABIElements.swift

```diff
@@ -264,72 +264,75 @@ extension ABI.Element.Function {
-            let check00_31 = BigUInt("08C379A000000000000000000000000000000000000000000000000000000000", radix: 16)!
-            let check32_63 = BigUInt("0000002000000000000000000000000000000000000000000000000000000000", radix: 16)!
-
-            // check data[00-31] and data[32-63]
-            if check00_31 == BigUInt(data[0...31]) && check32_63 == BigUInt(data[32...63]) {
-                // data.bytes[64-67] contains the length of require message
-                let len = (Int(data.bytes[64])<<24) | (Int(data.bytes[65])<<16) | (Int(data.bytes[66])<<8) | Int(data.bytes[67])
-
-                let message = String(bytes: data.bytes[68..<(68+len)], encoding: .utf8)!
-
-                print("read function aborted by require statement: \(message)")
-
-                var returnArray = [String: Any]()
-
-                // set infomation
-                returnArray["_abortedByRequire"] = true
-                returnArray["_errorMessageFromRequire"] = message
-
-                // set empty values
-                for i in 0 ..< outputs.count {
-                    let name = "\(i)"
-                    returnArray[name] = outputs[i].type.emptyValue
-                    if outputs[i].name != "" {
-                        returnArray[outputs[i].name] = outputs[i].type.emptyValue
-                    }
-                }
+    /// Decodes data returned by a function call. Able to decode `revert("...")` calls.
+    /// - Parameter data: bytes returned by a function call.
+    /// - Returns: a dictionary containing returned data mappend to indices and names of returned values if these are not `nil`.
+    /// Return cases:
+    ///     - when no `outputs` declared: returning `["_success": true]`;
+    ///     - when `outputs` declared and decoding completed successfully: returning `["_success": true, "0": value_1, "1": value_2, ...]`.
+    ///     Additionally this dictionary will have mappings to output names if these names are specified in the ABI.
+    ///     - function call was aborted using `require(some_string_error_message)`: returning `["_success": false, "_abortedByRequire": true, "_errorMessageFromRequire": error_message]`.
+    ///     - in case of any error: returning `["_success": false, "_failureReason": String]`;
+    ///         Error reasons include:
+    ///            -  `outputs` declared but at least one value failed to be decoded;
+    ///            - `data.count` is less than `outputs.count * 32`;
+    ///            - `outputs` defined and `data` is empty;
+    ///            - `data` represent reverted transaction
+    public func decodeReturnData(_ data: Data) -> [String: Any] {
+        guard !outputs.isEmpty else {
+            NSLog("Function doesn't have any output types to decode given data.")
+            return ["_success": true]
+        }
 
-                return returnArray
-            }
+        /// If data is empty and outputs are expected it is treated as a `requite(expression)` call with no message.
```

Comment:
minor typo: should be `require(expression)`

Reply:
Thanks!